src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py

```diff
@@ -188,941 +187,1485 @@
+    current_emissions_tasks = {}
+    prior_net_sequestration_tasks = {}
+    current_net_sequestration_tasks = {}
+    valuation_tasks = {}
+
+    first_transition_year = min(transition_years)
+    final_year = int(args['analysis_year'])
+
+    summary_net_sequestration_tasks = []
+    summary_net_sequestration_raster_paths = {
+        first_transition_year: args['sequestration_since_baseline_raster']}
+
+    for year in range(first_transition_year, final_year+1):
+        current_stock_tasks = {}
+        net_sequestration_rasters[year] = {}
+        stock_rasters[year] = {}
+        disturbance_vol_rasters[year] = {}
+        emissions_rasters[year] = {}
+        year_of_disturbance_rasters[year] = {}
+        valuation_tasks[year] = {}
+
+        for pool in (POOL_SOIL, POOL_BIOMASS):
+            # Calculate stocks from last year's stock plus last year's net
+            # sequestration.
+            stock_rasters[year][pool] = os.path.join(
+                intermediate_dir,
+                STOCKS_RASTER_PATTERN.format(
+                    year=year, pool=pool, suffix=suffix))
+            if year == first_transition_year:
+                current_stock_dependent_tasks = []
+                current_disturbance_vol_dependent_tasks = []
+            else:
+                current_stock_dependent_tasks = [
+                    prior_stock_tasks[pool],
+                    prior_net_sequestration_tasks[pool]]
+                current_disturbance_vol_dependent_tasks = [
+                    prior_stock_tasks[pool]]
+
+            current_stock_tasks[pool] = task_graph.add_task(
+                func=_sum_n_rasters,
+                args=([stock_rasters[year-1][pool],
+                       net_sequestration_rasters[year-1][pool]],
+                      stock_rasters[year][pool]),
+                dependent_task_list=current_stock_dependent_tasks,
+                target_path_list=[stock_rasters[year][pool]],
+                task_name=f'Calculating {pool} carbon stock for {year}')
+
+            # Calculate disturbance volume if we're at a transition year.
+            if year in transition_years:
+                # We should only switch around the transition years the first
+                # time we encounter this, not for each pool.
+                if current_transition_year != year:
+                    prior_transition_year = current_transition_year
+                    current_transition_year = year
+
+                disturbance_vol_rasters[year][pool] = os.path.join(
+                    intermediate_dir,
+                    DISTURBANCE_VOL_RASTER_PATTERN.format(
+                        pool=pool, year=year, suffix=suffix))
+                current_disturbance_vol_tasks[pool] = task_graph.add_task(
+                    func=pygeoprocessing.raster_calculator,
+                    args=([(disturbance_magnitude_rasters[year][pool], 1),
+                           # TODO: why is this using prior year's stocks?
```

Comment:
Is it okay to leave this TODO for a future issue?

Reply:
I've convinced myself that the emissions calculations as stated are using the correct carbon stocks raster, so c5451cb6 changes this from a `TODO` to a block comment describing my rationale for basing disturbance volume on last year's stocks.

Reply:
Actually, I take that back (and reverted c5451c b6).  effcb738 now includes the actual fix, which is consistent with the way that the model handles stocks elsewhere in the model.